uitools/toolkitwidgets.cmake

```diff
@@ -0,0 +1,66 @@
+# Copyright 2012-2024 Esri
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+include(${CMAKE_CURRENT_LIST_DIR}/common.cmake)
+
+set(WIDGETPATH ${CMAKE_CURRENT_LIST_DIR}/widgets/Esri/ArcGISRuntime/Toolkit)
+
+set(TOOLKITWIDGET_SOURCES
+    ${WIDGETPATH}/AuthenticationView.cpp
+    ${WIDGETPATH}/BasemapGallery.cpp
+    ${WIDGETPATH}/BookmarksView.cpp
+    ${WIDGETPATH}/CoordinateConversion.cpp
+    ${WIDGETPATH}/FloorFilter.cpp
+    ${WIDGETPATH}/NorthArrow.cpp
+    ${WIDGETPATH}/OverviewMap.cpp
+    ${WIDGETPATH}/Internal/ClientCertificatePasswordDialog.cpp
+    ${WIDGETPATH}/Internal/ClientCertificateView.cpp
+    ${WIDGETPATH}/Internal/CoordinateEditDelegate.cpp
+    ${WIDGETPATH}/Internal/Flash.cpp
+    ${WIDGETPATH}/Internal/OAuth2View.cpp
+    ${WIDGETPATH}/Internal/SslHandshakeView.cpp
+    ${WIDGETPATH}/Internal/UserCredentialView.cpp)
+
+set(TOOLKITWIDGET_RESOURCES ${CMAKE_CURRENT_LIST_DIR}/images/esri_arcgisruntime_toolkit_images.qrc)
+
+set(TOOLKITWIDGET_FORMS
+    ${WIDGETPATH}/AuthenticationView.ui
+    ${WIDGETPATH}/BasemapGallery.ui
+    ${WIDGETPATH}/BookmarksView.ui
+    ${WIDGETPATH}/CoordinateConversion.ui
+    ${WIDGETPATH}/FloorFilter.ui
+    ${WIDGETPATH}/OverviewMap.ui
+    ${WIDGETPATH}/Internal/ClientCertificatePasswordDialog.ui
+    ${WIDGETPATH}/Internal/ClientCertificateView.ui
+    ${WIDGETPATH}/Internal/OAuth2View.ui
+    ${WIDGETPATH}/Internal/SslHandshakeView.ui
+    ${WIDGETPATH}/Internal/UserCredentialView.ui)
+
+target_include_directories(${PROJECT_NAME} PRIVATE 
+    ${CMAKE_CURRENT_LIST_DIR}/widgets
+    ${WIDGETPATH})
+
+target_sources(${PROJECT_NAME} PRIVATE
+    ${TOOLKITCOMMON_SOURCES}
+    ${TOOLKITWIDGET_SOURCES}
+    ${TOOLKITWIDGET_RESOURCES}
+    ${TOOLKITWIDGET_FORMS})
```

Comment:
It's been a while since I worked with Widgets, but I am surprised that the .ui files would appear in the `target_sources` list. 
This is a separate list in qmake via `FORMS`. I assume it's tested, but it seems different.

Reply:
I agree it is strange, but this is how Qt has it on their page: https://doc.qt.io/qt-6/designer-using-a-ui-file.html#compile-time-form-processing

specifically for our widgets:
> The form is listed among the C++ source files in qt_add_executable(). The option CMAKE_AUTOUIC tells CMake to run the uic tool to create a ui_calculatorform.h file that can be used by the source files.

Reply:
Good to know, thanks.